upgrades.js

```diff
@@ -26,6 +26,19 @@ H5PUpgrades['H5P.Flashcards'] = (function () {
 
         // Done
         finished(null, parameters, extras);
+      },
+      7: function (parameters, finished, extras) {
+
+        if (params && Array.isArray(parameters.cards)) {
+          for (var i = 0; i < parameters.cards.length; i++) {
+            if (typeof parameters.cards[i].answer === 'string') {
+              parameters.cards[i].answer = parameters.cards[i].answer
+                .replace(/\|/g, '/')
+                .replace(/\\\|/g, '\\/')
```

Comment:
Can this rule hit if all pipe symbols are replaced in the previous line?

Reply:
Sorry, just had done this quickly on my mobile at breakfast :-D

I fear you will need to take care of a little more. All symbols involved need to be taken into account, so
- `|` for the "logical or" needs to become a `/`,
- `\|` for a pipe symbol needs to become a `|`, and
- `/` for a slash symbol needs to become a `\/`.

A test string that had everything in it could be `foo|bar\|batz/yada` where the typed solutions for the fields are `foo` and `bar|batz/yada`.

Your code would result in `foo\/bar\\/batz\/yada` and yield `foo/bar\/batz/yada` as the only expected solution.

A common approach is to replace symbols greedily with an intermediate placeholder that is guaranteed to not be (or at least unlikely to be) present in the text, so subsequent replacements do no revert any change so far. And then one changes the placeholders into what they are supposed to be ultimately, e.g. like

```
parameters.cards[i].answer = parameters.cards[i].answer
  .replace(/\\\|/g, '__TO_BE_AN_UNESCAPED_PIPE__') // Replace escaped pipe symbol
  .replace(/\|/g, '__TO_BE_A_LOGICAL_OR_AKA_SLASH__') // Replace pipe symbol
  .replace(/\//g, '__TO_BE_AN_ESCAPED_SLASH__') // Replace slash
  .replace(/__TO_BE_AN_UNESCAPED_PIPE__/g, '\|')
  .replace(/__TO_BE_A_LOGICAL_OR_AKA_SLASH__/g, '/')
  .replace(/__TO_BE_AN_ESCAPED_SLASH__/g, '\\/');
```

The placeholders like `__TO_BE_AN_UNESCAPED_PIPE__` could be something else, of course, but they are quite expressive and unlikely to be in the original solution :-D

Please also note that your code would crash (because of `params` instead of `parameters` in line 32) and the upgrade process would never end, because you're not calling the `finished` callback.